programs/bpf_loader/src/syscalls.rs

```diff
@@ -924,29 +989,46 @@ mod tests {
     #[should_panic(expected = "UserError(SyscallError(Panic(\"Gaggablaghblagh!\", 42, 84)))")]
     fn test_syscall_sol_panic() {
         let string = "Gaggablaghblagh!";
-        let addr = string.as_ptr() as *const _ as u64;
-        let ro_region = MemoryRegion {
-            addr_host: addr,
-            addr_vm: 100,
-            len: string.len() as u64,
+        solana_logger::setup_with_default("solana=info");
+
+        let mut mock_invoke_context = MockInvokeContext::default();
+        let result = {
+            let invoke_context: &mut dyn InvokeContext = &mut mock_invoke_context;
+            //let invoke_context: &mut dyn InvokeContext = &mut MockInvokeContext::default();
+            let mut syscall_sol_panic = SyscallPanic {
+                invoke_context: Rc::new(RefCell::new(invoke_context)),
+            };
+
+            let addr = string.as_ptr() as *const _ as u64;
+            let ro_region = MemoryRegion {
+                addr_host: addr,
+                addr_vm: 100,
+                len: string.len() as u64,
+            };
+            let rw_region = MemoryRegion::default();
+            syscall_sol_panic.call(
+                100,
+                string.len() as u64,
+                42,
+                84,
+                0,
+                &[ro_region],
+                &[rw_region],
+            )
         };
-        let rw_region = MemoryRegion::default();
-        syscall_sol_panic(
-            100,
-            string.len() as u64,
-            42,
-            84,
-            0,
-            &[ro_region],
-            &[rw_region],
-        )
-        .unwrap();
+        assert_eq!(mock_invoke_context.log.len(), 1);
+        assert_eq!(
+            mock_invoke_context.log[0],
+            "panicked at \'Gaggablaghblagh!\' on 42:84"
+        );
+
+        result.unwrap();
     }
 
     // Ignore this test: solana_logger conflicts when running tests concurrently,
     // this results in the bad string length being ignored and not returning an error
     #[test]
-    #[ignore]
+    #[serial]
```

Comment:
I tried serial for these two tests and it appeared to work but then started showing up in CI as failures...

Reply:
Just noticed that CI failed on this test as expected due to conflict.  Commited a change to ignore these tests again to unblock CI

Reply:
I just reworked these tests to avoid using `solana_logger::setup_with_default("solana=info");` entirely.  With the MockInvokeContext, we can actually collect and validate that the log functions actually output the expected result insteadl

Reply:
Strange that it fails on CI but not locally, maybe the conflict is not what I thought it was

Reply:
Maybe I'm reading it wrong but doesn't MockInvokeContext still use solana_logger to determine if the log message is enabled?

Reply:
Sry I'm moving the furniture on you.

MockInvokeContext:
```rust
        fn log_enabled(&self) -> bool {
            true
        }
        fn log(&mut self, message: &str) {
            info!("[MockInvokeContext::log] {}", message);
            self.log.push(message.to_string());
        }
```

ThisInvokeContext:
```rust
    fn log_enabled(&self) -> bool {
        log_enabled!(log::Level::Info) || self.log_collector.is_some()
    }

    fn log(&mut self, message: &str) {
        info!("{}", message);
        if let Some(log_collector) = self.log_collector {
            log_collector.log(message);
        }
    }
```